src/app/services/slices/query-status.slice.ts

```diff
@@ -1,19 +1,28 @@
 import { createSlice, PayloadAction } from '@reduxjs/toolkit';
-import { IStatus } from '../../../types/status';
-
-type Status = Partial<IStatus>
 
-const initialState: Status = {}
+import { IStatus } from '../../../types/status';
+import { LOGOUT_SUCCESS, QUERY_GRAPH_RUNNING } from '../redux-constants';
 
-const queryStatus = createSlice({
-  name: 'queryStatus',
-  initialState,
+const queryRunnerStatusSlice = createSlice({
+  name: 'queryRunnerStatus',
+  initialState: null as IStatus | null,

```

Comment:
Hmm... I wonder why you're casting null to `Istatus`? Is it because `initialState` can have fields with `null` values?

Reply:
yes.

Reply:
and it was the only way to keep the compiler from shouting at me